pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -668,10 +668,24 @@ impl Timeline {
 
         // Define partitioning schema if needed
 
+        let l0_l1_boundary_lsn = {
+            // We do the repartition on the L0-L1 boundary. All data below the boundary
+            // are compacted by L0 with low read amplification, thus making the `repartition`
+            // function run fast.
```

Comment:
Do we really need this change now that we don't do L1 compaction until L0 is compacted down, thus bounding the L0 read amp at 10?

I think the change is reasonable in any case -- if the data set is large enough to care about L1 churn then L0 will be small compared to L1. I'm just not sure we really need it anymore.

Reply:
On second thought, maybe we should do this after all. Even if read amp is low at ~10, it's still a win to avoid those layers.